Tools/px_uploader.py

```diff
@@ -441,9 +459,14 @@ def __erase(self, label):
 
             if self.__trySync():
                 self.__drawProgressBar(label, 10.0, 10.0)
+                if self.force_erase:
+                    print("\nForce erase done.\n")
                 return
 
-        raise RuntimeError("timed out waiting for erase")
+        if self.force_erase:
+            raise RuntimeError("timed out waiting for erase, force erase is likely not supported by bootloader!")
```

Comment:
Wouldn't the the force erase bet met with an immediate command error? In that case should you attempt to use an erase or suggest a bootloader upgrade?

Reply:
Yes, that would be nice. But a unknown byte seems to just be ignored. And the erase command is not acked until it is erased, so you have to wait a long time. Not optimal I know...

Reply:
geez That seems wrong, bit it is what the code does.

Reply:
Yep.